src/components/generic/movieModal/modalContent.tsx

```diff
@@ -5,55 +5,59 @@ import {x} from 'react-icons-kit/feather'
 import {twitter} from 'react-icons-kit/icomoon/twitter'
 import {facebook} from 'react-icons-kit/fa'
 import {PoseGroup} from 'react-pose'
-import SelectQuality from './SelectQuality'
-import useLockBodyScroll from '../../hooks/useLockBodyScroll'
-import {Props} from './MovieModal'
 
 import * as Modal from './style'
+import {useModal} from '@/context/modal-context'
+import {useMovie} from '@/api/movie_api'
 
-const Category: FC<{title: string; children: React.ReactChild}> = ({
-  title,
-  children,
-}) => (
-  <Modal.Category>
-    <Modal.CatTitle>{title}</Modal.CatTitle>
-    {children}
-  </Modal.Category>
-)
-
-const ModalContent: FC<Props> = ({
-  open,
+export interface Data {
+  overview: string
+  genres: genre[]
+  id: string
+  poster_path: string
+  vote_average: number
+  title: string
+  release_date: number
+}
 
-  handleClose,
-  data: {
-    title,
-    poster_big,
-    rating,
-    description,
-    year,
-    genres,
-    trailer,
-    items,
-    imdb,
-  },
-}) => {
+export const ModalContent: FC = () => {
   //Fix me:lockscroll
   // {
   //   open ? useLockBodyScroll('lock') : useLockBodyScroll('unlock')
   // }
-  return (
+
+  const {movieId} = useModal()
+  const {
+    movie,
+    isFetching,
+    isIdle,
+    isSuccess,
+  }: {
+    movie: Data
+    isFetching: boolean
+    isIdle: boolean
+    isSuccess: boolean
+  } = useMovie(movieId)
+  const {isOpen, setIsOpen} = useModal()
+
+  return isIdle ? null : isFetching ? (
```

Comment:
Nested ternaries are harder to read. I suggest doing explicit returns even if you write more code.

Reply:
Done.